### PR TITLE
cargo-dist

### LINKED ISSUE #415: Distribute binaries with each release
**Is your feature request related to a problem? Please describe.**

In order to improve the ease of use/installation of Topiary, I believe it would be good to provide compiled binaries for common architecture with each release. This would allow random users to just download and run Topiary.

More importantly, this would open the door to easy distribution as a package, because this would not require any more compilation from the package maintainers. This could allow distribution through OPAM, for instance, or alternative faster distribution through nixpkgs (which, for now, contains only the version of Topiary one needs to compile).

These considerations came up in the context of https://github.com/tweag/topiary/issues/254.

### PR DESCRIPTION
**Pull request by torhovland**
Fixes #415.

CI will automatically cut a release like this when pushing a new version tag.

image

### CODE REVIEW

File: CHANGELOG.md
Comment (Xophmeister) [inline CHANGELOG.md:51]: By removing the links to the GitHub comparison, these links are no longer referenced from anywhere, so they're redundant. Do we want to keep exposing them somehow?

(n.b., It looks like the markdown link tag for the 0.2.3 release isn't correct, here, but the general point still stands.)

File: CHANGELOG.md
Comment (Xophmeister) [inline CHANGELOG.md:47]: It seems a shame to lose the release date.

**Review by Xophmeister — COMMENTED** [review]
This is cool :+1: I've just added a few comments on what could be seen as potential regressions to the changelog format. I assume this is to conform with `cargo-dist`'s expectations...

File: MAINTAINERS.md
Comment (ErinvanderVeen) [inline MAINTAINERS.md:79]:
```suggestion
[changelog]: /CHANGELOG.md
```

**Review by ErinvanderVeen — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (ErinvanderVeen) [reply CHANGELOG.md:47]: I agree with this. While the GitHub release do have a date associated with them, I would like the repo to be able to stand on it's own.

File: CHANGELOG.md
Comment (ErinvanderVeen) [reply CHANGELOG.md:51]: I don't mind this too much actually. The links are still there for those editing the file.

Comment (torhovland) [reply CHANGELOG.md:51]: We should be able to add a link back, as long as the braces aren't around the version number.

File: CHANGELOG.md
Comment (torhovland) [reply CHANGELOG.md:47]: We can put the dates back in, they'll just be slightly redundant on the GitHub release page.

Comment (ErinvanderVeen) [reply CHANGELOG.md:47]: Either way, I'm happy.

**Review by ErinvanderVeen — APPROVED** [review]

File: Cargo.toml
Comment (ErinvanderVeen) [reply Cargo.toml:35]: Oh, yeah, definitely, that would also include the Raspberry Pi, and some NixOS mobile devices.

Comment (torhovland) [reply Cargo.toml:35]: It's just that the above was what `cargo-dist` did by default. There may be a reason for that. Anyway, I will try.

Comment (torhovland) [reply Cargo.toml:35]: > In v0.0.5 the only properly supported choices are:
> x64 macOS: "x86_64-apple-darwin"
> x64 Windows: "x86_64-pc-windows-msvc"
x64 Linux: "x86_64-unknown-linux-gnu"
arm64 macOS (Apple silicon): "aarch64-apple-darwin" (supported added in v0.0.4)
Future versions should hopefully introduce proper support for important targets like "musl linux".

Comment (torhovland) [reply Cargo.toml:35]: I opened https://github.com/tweag/topiary/issues/540

File: CHANGELOG.md
Comment (torhovland) [reply CHANGELOG.md]: This has been improved now.

File: CHANGELOG.md
Comment (torhovland) [reply CHANGELOG.md:47]: Fixed.

**Comment by torhovland** [comment]
Another test release:

image

**Comment by torhovland** [comment]
I feel I need one more (re-)approval before merging, please.

**Comment by torhovland** [comment]
> Another test release:

In the normal case, the release title would be "v0.2.4 - Cyclic Cypress - 2023-06-29"

**Review by ErinvanderVeen — APPROVED** [review]

**Pull request merged**
